### PR TITLE
Move VPack to OneBranch

### PR DESCRIPTION
**Pull request by csujedihy**
## Description

Move VPack to OneBranch pipeline.

## Testing

Tested in OneBranch pipeline

### CODE REVIEW

File: .azure/OneBranch.Package.yml [hunk 1]
```diff
@@ -5,8 +5,17 @@ resources:
...
+    - repository: templates
+      type: git
+      name: OneBranch.Pipelines/GovernedTemplates
+      ref: refs/heads/main
 
 parameters:
+- name: 'debug'
+  displayName: 'Enable debug output'
+  type: boolean
+  default: false
```
Comment (nibanks) [inline .azure/OneBranch.Package.yml:18]: Not actually needed. We never use it.

File: .azure/OneBranch.Package.yml [hunk 2]
```diff
@@ -23,111 +32,101 @@ parameters:
 name: 0.$(Date:yyyy).$(Date:MM).$(DayOfMonth).$(Rev:rr).0
 
 variables:
+  CDP_DEFINITION_BUILD_COUNT: $[counter('', 0)] # needed for onebranch.pipeline.version task https://aka.ms/obpipelines/versioning
+  system.debug: ${{ parameters.debug }}
+  ENABLE_PRS_DELAYSIGN: 1
```
Comment (nibanks) [inline .azure/OneBranch.Package.yml:37]: Also not needed.

File: .azure/OneBranch.Package.yml [hunk 3]
```diff
@@ -23,111 +32,101 @@ parameters:
...
-  - push_chpefre
-  - push_chpechk
-  - push_x86fre
-  - push_x86chk
-  jobs:
-  - template: ./obtemplates/git-submit.yml
+  ROOT: $(Build.SourcesDirectory)
+  REPOROOT: $(Build.SourcesDirectory)
+  OUTPUTROOT: $(REPOROOT)\out
+  NUGET_XMLDOC_MODE: none
```
Comment (nibanks) [inline .azure/OneBranch.Package.yml:42]: None of this is needed either.

File: .azure/OneBranch.Package.yml [hunk 4]
```diff
@@ -23,111 +32,101 @@ parameters:
...
+      enabled: false
+    globalSdl: # https://aka.ms/obpipelines/sdl
+      tsa:
+        enabled: false # onebranch publish all sdl results to TSA. If TSA is disabled all SDL tools will forced into 'break' build mode.
+      credscan:
+        suppressionsFile: $(Build.SourcesDirectory)\.azure\CredScanSuppressions.json
+      binskim:
+        break: true # always break the build on binskim issues in addition to TSA upload
+      policheck:
+        break: true # always break the build on policheck issues. You can disable it by setting to 'false'
```
Comment (nibanks) [inline .azure/OneBranch.Package.yml:60]: I believe this are actually the defaults, so they aren't needed either.

File: .azure/OneBranch.Package.yml [hunk 5]
```diff
@@ -23,111 +32,101 @@ parameters:
...
+  NUGET_XMLDOC_MODE: none
+  WindowsContainerImage: 'onebranch.azurecr.io/windows/ltsc2022/vse2022:latest'
+
+extends:
+  template: v2/Microsoft.Official.yml@templates # https://aka.ms/obpipelines/templates
+  parameters:
+    platform:
+      name: 'windows_undocked' # windows undocked
+    cloudvault: # https://aka.ms/obpipelines/cloudvault
+      enabled: false
```
Comment (nibanks) [inline .azure/OneBranch.Package.yml:51]: Does this pipeline actually use this?

File: .azure/OneBranch.Package.yml [hunk 6]
```diff
@@ -23,111 +32,101 @@ parameters:
...
+          group: VPackDeploymentKeys
+          ob_outputDirectory: $(Build.SourcesDirectory)/artifacts/package/$(platform)
+          ob_artifactSuffix: _$(platform)
+          ob_sdl_codeSignValidation_excludes: -|**\*.sys;-|**\*.dll;-|**\*.exe
+          ob_createvpack_enabled: true
+          ob_createvpack_packagename: msquic.$(platform)
+          ob_createvpack_owneralias: quicdev
+          ob_createvpack_description: msquic.$(Build.SourceBranchName)
+          ob_createvpack_versionAs: string
+          ob_createvpack_version: 2.3.0-$(Build.BuildId)
```
Comment (nibanks) [inline .azure/OneBranch.Package.yml:115]: Does our update version ps1 script still work for this, or does it need to be updated?

File: .azure/OneBranch.Package.yml [hunk 7]
```diff
@@ -23,111 +32,101 @@ parameters:
...
+    - stage: push_git
+      displayName: Push to Git
+      dependsOn:
+      - build_vpack
+      jobs:
+      - template: .azure/obtemplates/git-submit.yml@self
+        parameters:
+          branch: ${{ parameters.branch }}
+          title: ${{ parameters.title }}
+          #msrc: ${{ parameters.msrc }}
```
Comment (nibanks) [inline .azure/OneBranch.Package.yml:131]: Please add a trailing newline.

File: .azure/obtemplates/git-submit.yml [hunk 8]
```diff
@@ -1,24 +1,36 @@
...
-  - group: VPackDeploymentKeys
+    ob_outputDirectory: $(Build.SourcesDirectory)/artifacts/package
+    ob_sdl_codeSignValidation_excludes: -|**\*.sys;-|**\*.dll;-|**\*.exe
+    runCodesignValidationInjection: false
+    # Update OS Manifest parameters - Settings for checking in manifest to OS repo.
+    ob_updateOSManifest_enabled: true
+    ob_updateOSManifest_osbranch: 'official/rs_onecore_liof1_stack'
+    ob_updateOSManifest_gitcheckinConfigPath: '$(Build.SourcesDirectory)/artifacts/package/GitCheckin.json'
+    ob_updateOSManifest_source: '/minio/netio/quic/msquic/msquic.man'
+    ob_updateOSManifest_destination: '$(Build.SourcesDirectory)/artifacts/package'
```
Comment (nibanks) [inline .azure/obtemplates/git-submit.yml:22]: Branch, source and destination are already encoded in the json I believe and are unnecessary to specify here. i didn't need to specify them in my [build.yml](https://mscodehub.visualstudio.com/undocking/_git/undocking?path=/onebranch/v2/build.yml&version=GBmain&line=161&lineEnd=164&lineStartColumn=1&lineEndColumn=73&lineStyle=plain&_a=contents).

File: .azure/OneBranch.Package.yml (hunk 4 above)
Comment (csujedihy) [reply .azure/OneBranch.Package.yml:60]: Are you sure credscan and supppressionsFile are also the defaults?

File: .azure/obtemplates/git-submit.yml (hunk 8 above)
Comment (csujedihy) [reply .azure/obtemplates/git-submit.yml:22]: Oh right. Though, destination will still be needed.

File: .azure/OneBranch.Package.yml (hunk 5 above)
Comment (csujedihy) [reply .azure/OneBranch.Package.yml:51]: nope

File: .azure/OneBranch.Package.yml (hunk 6 above)
Comment (csujedihy) [reply .azure/OneBranch.Package.yml:98]: Actually, there is an issue. The BuildId in vpack version won't match  MSQUIC_METADATA_ONEBRANCH_BUILD_ID.

Comment (csujedihy) [reply .azure/OneBranch.Package.yml:98]: MSQUIC_METADATA_ONEBRANCH_BUILD_ID won't match the build ID of the pipeline that pushes vpack and creates the PR.

File: .azure/OneBranch.Package.yml (hunk 4 above)
Comment (csujedihy) [reply .azure/OneBranch.Package.yml]: I saw you highlighted only 4 lines. So, they are removed now.

File: .azure/OneBranch.Package.yml (hunk 6 above)
Comment (csujedihy) [reply .azure/OneBranch.Package.yml:98]: Looks like we have been doing all the time. So, probably not a big deal.

Comment (csujedihy) [reply .azure/OneBranch.Package.yml:98]: >> Does our update version ps1 script still work for this, or does it need to be updated?
nvm. I updated update-version.ps1.

**Review by nibanks — APPROVED** [review]

**Pull request merged**